### PR TITLE
Request promise to got

### PR DESCRIPTION
**Pull request by Delodax**
So, some notes on what I've done thus far:

- I removed the "default pagination value", since I found it too cumbersome adapting to potential input from the user. AFAIK it can be an object, an instace of `URLSearchParams` or even a string. Thus I simply swallow whatever the user inputs and roll with it.
- I made some slight alterations to the README to reflect my changes - please inspect!

### CODE REVIEW

**Review by exacs — COMMENTED** [review]
I want to check that passing an array as query parameters still work with this update.

The new suggested way (as written in the README) of doing it doesn't look very obvious:

```js
const pages = await canvas.listPaginated(`/courses/${courseId}/users`, new URLSearchParams([['include[]', 'avatar_url'], ['include[]', 'uuid']]))
```

I would prefer to have some kind of support for things like this even if it means that we are limiting the original `got` interface…

```js
const pages = await canvas.listPaginated(`/courses/${courseId}/users`, {include: ['avatar_url', 'uuid']})
```

Also –at least, in previous versions– the Canvas API didn't accept all array formats for query parameters and would be good that this package formats the array correctly so the users of it don't need to know about specific details from Canvas…

**Comment by Delodax** [comment]
I will explore using "baseUrl" and restrict the "get" to only accept object (and run it through `query-string`.

Also, I'll reintroduce the default pagination of 100.

File: index.js [hunk 1]
```diff
@@ -18,43 +18,43 @@ function getNextUrl (linkHeader) {
 module.exports = (apiUrl, apiKey, options = {}) => {
   const log = options.log || (() => {})
 
-  const resolve = endpoint => {
-    const base2 = apiUrl.slice(-1) === '/' ? apiUrl.slice(0, -1) : apiUrl
-    const endpoint2 = endpoint.charAt(0) === '/' ? endpoint.slice(1) : endpoint
+  const canvasGot = got.extend({
+    headers: {
```
Comment (Delodax) [inline index.js:22]: I was thinking of moving the `baseUrl` into here, since we use if every time - whaddya say?

File: README.md [hunk 2]
```diff
@@ -9,7 +9,7 @@ NodeJS HTTP client based on [Request-Promise](https://github.com/request/request
 
 Once you build a canvas instance, you get an object with only four methods:
 
-1. The low-level `requestUrl()` method to perform any request with any HTTP method
+1. The low-level `requestUrl()` method to perform a request with any HTTP method
 2. `get()` to perform a GET request
```
Comment (Delodax) [inline README.md:13]: Was thinking about adding some comments for `get` and the `list`/`listPaginated` below, noting how to supply a query string - whaddya think?

**Review by Delodax — COMMENTED** [review]
Two comments to ponder! 😄
@exacs

File: index.js (hunk 1 above)
Comment (Delodax) [reply index.js:22]: Let's not - due to pagination logic.

File: README.md (hunk 2 above)
Comment (Delodax) [reply README.md:13]: Added some new text to the README!

**Pull request merged**